icu4j/main/classes/core/src/com/ibm/icu/number/LocalizedNumberFormatter.java

```diff
@@ -101,7 +102,7 @@ public FormattedNumber format(Measure input) {
         MeasureUnit unit = input.getUnit();
         FormattedStringBuilder string = new FormattedStringBuilder();
         formatImpl(fq, unit, string);
-        return new FormattedNumber(string, fq);
+        return new FormattedNumber(string, fq, input.getUnit());
```

Comment:
Wrong unit.  You need to return the one from MicroProps, which is hidden inside formatImpl.  You need to get it out of formatImpl.

Please add a test for this.

Reply:
cone